core/templates/dev/head/pages/exploration-editor-page/translation-tab/audio-translation-bar/audio-translation-bar.directive.ts

```diff
@@ -302,32 +303,34 @@ oppia.directive('audioTranslationBar', [
           $scope.saveRecordedAudio = function() {
             $scope.audioIsCurrentlyBeingSaved = true;
             SiteAnalyticsService.registerSaveRecordedAudioEvent();
-            var filename = generateNewFilename();
-            var fileType = 'audio/mp3';
-            var contentId = $scope.contentId;
-            var languageCode = $scope.languageCode;
-            var recordedAudioFile = new File(
-              [$scope.audioBlob], filename, {type: fileType});
-            $scope.showRecorderWarning = false;
-            AssetsBackendApiService.saveAudio(
-              ContextService.getExplorationId(), filename,
-              recordedAudioFile).then(function() {
-              if ($scope.audioIsUpdating) {
-                StateRecordedVoiceoversService.displayed.deleteVoiceover(
-                  contentId, languageCode);
-                $scope.audioIsUpdating = false;
-              }
-              StateRecordedVoiceoversService.displayed.addVoiceover(
-                contentId, languageCode, filename, recordedAudioFile.size);
-              saveRecordedVoiceoversChanges();
-              AlertsService.addSuccessMessage(
-                'Succesfuly uploaded recorded audio.');
-              $scope.audioIsCurrentlyBeingSaved = false;
-              $scope.initAudioBar();
-            }, function(errorResponse) {
-              $scope.audioIsCurrentlyBeingSaved = false;
-              AlertsService.addWarning(errorResponse.error);
-              $scope.initAudioBar();
+            AssetsBackendApiService.initialize().then(function() {
```

Comment:
I'm not sure why we have to initialize the service every time, although I've initialized it in the begging. Can we move the `GCS_RESOURCE_BUCKET_NAME` check inside the `saveAudio` method?